agent/agent.go

```diff
@@ -235,14 +236,16 @@ func (a *agent) TailnetConn() *tailnet.Conn {
 }
 
 func (a *agent) init(ctx context.Context) {
-	sshSrv, err := agentssh.NewServer(ctx, a.logger.Named("ssh-server"), a.prometheusRegistry, a.filesystem, a.sshMaxTimeout, "")
+	sshSrv, err := agentssh.NewServer(ctx, a.logger.Named("ssh-server"), a.prometheusRegistry, a.filesystem, &agentssh.Config{
+		MaxTimeout:       a.sshMaxTimeout,
+		MOTDFile:         func() string { return a.manifest.Load().MOTDFile },
+		ServiceBanner:    func() *codersdk.ServiceBannerConfig { return a.serviceBanner.Load() },
+		UpdateEnv:        a.updateCommandEnv,
+		WorkingDirectory: func() string { return a.manifest.Load().Directory },
```

Comment:
Looking at how `agentssh.Config` is used here, would it make more sense to pass `Manifest func() Manifest` instead of MOTDFile, ServiceBanner, WorkingDirectory etc. ?

Reply:
This introduces a strong coupling between `agentsdk` and `agentssh` server, I would like us to keep it decoupled and think more about `agentssh` like `sshd`.